routers/user.py

```diff
@@ -0,0 +1,259 @@
+import httpx
+
+from auth.config import get_settings
+from auth.management import get_management_token
+from auth.validator import get_current_user
+from datetime import datetime, timezone
+from fastapi import APIRouter, Depends, HTTPException
+from schemas.requests import ResourceRequest, ServiceRequest
+from schemas.user import User
+
+router = APIRouter()
+
+def get_settings_instance():
+    return get_settings()
+
+async def fetch_user_data(user_id: str, token: str):
+    url = f"https://{get_settings_instance().auth0_domain}/api/v2/users/{user_id}"
+    headers = {"Authorization": f"Bearer {token}"}
+    async with httpx.AsyncClient() as client:
+        response = await client.get(url, headers=headers)
+        if response.status_code != 200:
+            raise HTTPException(
+                status_code=response.status_code,
+                detail="Failed to fetch user data"
+            )
+        return response.json()
+
+
+async def update_user_metadata(user_id: str, token: str, metadata: dict):
+    """Utility function to update user metadata in Auth0."""
+    url = f"https://{ get_settings_instance().auth0_domain}/api/v2/users/{user_id}"
+    headers = {
+        "Authorization": f"Bearer {token}",
+        "Content-Type": "application/json",
+    }
+    async with httpx.AsyncClient() as client:
+        response = await client.patch(
+            url,
+            headers=headers,
+            json={"app_metadata": metadata}
+        )
+        if response.status_code != 200:
+            raise HTTPException(
+                status_code=response.status_code,
+                detail="Failed to update user metadata"
+            )
+        return response.json()
+
+
+@router.get("/me/services")
+async def get_all_services(user: User = Depends(get_current_user)):
+    user_id = user.access_token.sub
+    token = get_management_token()
+    user_data = await fetch_user_data(user_id, token)
+    services = user_data.get("app_metadata", {}).get("services", [])
```

Comment:
This is the kind of thing that could be done more cleanly if we had a schema/model for the user data.